skfda/tests/test_fpca_regression.py

```diff
@@ -0,0 +1,212 @@
+"""Tests for FPCA."""
+from __future__ import annotations
+
+import unittest
+
+import numpy as np
+
+import skfda
+from skfda.datasets import fetch_tecator
+from skfda.misc.operators import LinearDifferentialOperator
+from skfda.misc.regularization import L2Regularization
+from skfda.ml.regression import FPCARegression
+from skfda.representation.basis import BSplineBasis
+
+
+class FPCARegressionTestCase(unittest.TestCase):
+    """Tests for principal component analysis."""
+
+    def test_fpca_reg_against_fda_usc(self) -> None:
+        """Check that the results obtained are similar to those of fda.usc.
+
+        Results obtained from fda.usc with the following R code
+            library("fda.usc")
```

Comment:
Were you able to compare results with refund too?

Reply:
This is the issue that I opened: https://github.com/refunders/refund/issues/102.
It doesn't seem like we'll be getting any solutions from their side anytime soon.